vnext/Desktop/Modules/WebSocketModule.cpp

```diff
@@ -148,27 +148,39 @@ std::shared_ptr<IWebSocketResource> WebSocketModule::GetOrCreateWebSocket(int64_
   {
     auto ws = IWebSocketResource::Make(std::move(url));
     auto weakInstance = this->getInstance();
-    ws->SetOnError([this, id, weakWs = weak_ptr<IWebSocketResource>(ws), instance = weakInstance.lock()](const IWebSocketResource::Error& err)
+    ws->SetOnError([this, id, weakInstance](const IWebSocketResource::Error& err)
     {
-      auto strongWs = weakWs.lock();
+      auto strongInstance = weakInstance.lock();
+      if (!strongInstance)
+        return;
+
       auto errorObj = dynamic::object("id", id)("message", err.Message);
       this->SendEvent("websocketFailed", std::move(errorObj));
```

Comment:
What's keeping your this pointer alive? Is the assumption that it is alive by virtue of strongInstance being alive? Or should we capture a weak ref to "this" as well?

Reply:
This `CxxModule` is ultimately held by a `vector<unique_ptr<CxxModule>>` container.
Can I create a `weak_ptr` to a raw pointer?

As you mentioned, the assumption of life time has been based on the owning `Instance` still being alive.